src/py/aspen/cli/db.py

```diff
@@ -108,3 +128,115 @@ def import_covidhub_project(
+)
+@click.option("--builds-template-args", type=str, required=True)
+@click.pass_context
+def create_phylo_run(
+    ctx,
+    group_name,
+    all_group_sequences,
+    samples,
+    aligned_gisaid_dump_id,
+    builds_template_file,
+    builds_template_args,
+):
+    # these are injected into the IPython scope, but they appear to be unused.
+    engine = ctx.obj["ENGINE"]
+
+    with session_scope(engine) as session:
+        group = session.query(Group).filter(Group.name == group_name).one()
+
+        all_samples: Iterable[Sample] = (
+            session.query(Sample)
+            .filter(
+                or_(
+                    Sample.public_identifier.in_(samples),
+                    and_(
+                        expression.true()
+                        if all_group_sequences
+                        else expression.false(),
+                        Sample.submitting_group == group,
+                    ),
+                )
+            )
+            .options(
+                joinedload(Sample.sequencing_reads_collection)
+                .joinedload(SequencingReadsCollection.consuming_workflows)
+                .joinedload(Workflow.outputs)
+                .joinedload(Entity.consuming_workflows)
+                .joinedload(Workflow.outputs)
+                .joinedload(Entity.consuming_workflows)
+                .joinedload(Workflow.outputs),
+                joinedload(Sample.uploaded_pathogen_genome),
+            )
+        )
+
+        pathogen_genomes: MutableSequence[PathogenGenome] = list()
+        for sample in all_samples:
+            if sample.sequencing_reads_collection is not None:
+                sample_pathogen_genomes = (
+                    sample.sequencing_reads_collection.pathogen_genomes
+                )
+                # TODO: We are blindly using the first pathogen genome we encounter.  It
+                # should probably be based on some heuristic.
+                if len(sample_pathogen_genomes) > 0:
+                    pathogen_genomes.append(sample_pathogen_genomes[0])
+            elif sample.uploaded_pathogen_genome is not None:
+                pathogen_genomes.append(sample.uploaded_pathogen_genome)
+            else:
+                raise ValueError(
+                    "Sample without sequencing reads collection nor uploaded"
+                    " pathogen"
+                )
+        if len(pathogen_genomes) == 0:
+            raise ValueError("No sequences selected for run.")
+
+        if aligned_gisaid_dump_id is not None:
+            aligned_gisaid_dump = (
+                session.query(AlignedGisaidDump)
+                .filter(AlignedGisaidDump.entity_id == aligned_gisaid_dump_id)
+                .one()
+            )
+        else:
+            aligned_gisaid_dump = (
+                session.query(AlignedGisaidDump)
+                .join(AlignedGisaidDump.producing_workflow)
+                .order_by(Workflow.end_datetime.desc())
```

Comment:
oh chill, so it just get's the most recent one?

Reply:
yup.